examples/todo/src/controllers/model-admin.controller.ts

```diff
@@ -0,0 +1,242 @@
+  // FIXME(bajtos) fix discoverSchemas, it does not return PromiseOrVoid
+  const discoveredModels = (await ds.discoverSchemas(table)) as AnyObject;
+  /* Example output from MySQL:
+     {
+       'test.CoffeeShop': {
+         name: 'Coffeeshop',
+         options: {
+           idInjection: false,
+           mysql: {schema: 'test', table: 'CoffeeShop'},
+         },
+         properties: {
+           id: {
+             type: 'Number',
+             required: true,
+             length: null,
+             precision: 10,
+             scale: 0,
+             id: 1,
+             mysql: {
+               columnName: 'id',
+               dataType: 'int',
+               dataLength: null,
+               dataPrecision: 10,
+               dataScale: 0,
+               nullable: 'N',
+             },
+           },
+           name: {
+             type: 'String',
+             required: false,
+             length: 512,
+             precision: null,
+             scale: null,
+             mysql: {
+               columnName: 'name',
+               dataType: 'varchar',
+               dataLength: 512,
+               dataPrecision: null,
+               dataScale: null,
+               nullable: 'Y',
+             },
+           },
+           city: {
+             type: 'String',
+             required: false,
+             length: 512,
+             precision: null,
+             scale: null,
+             mysql: {
+               columnName: 'city',
+               dataType: 'varchar',
+               dataLength: 512,
+               dataPrecision: null,
+               dataScale: null,
+               nullable: 'Y',
+             },
+           },
+       }}}
+     */
+  const jugglerDef = discoveredModels[Object.keys(discoveredModels)[0]];
+
+  return new ModelDefinition({
+    name: jugglerDef.name,
+    // TODO: convert from juggler/LB3 style to LB4
+    // For example, we need to transform array-type definitions.
+    properties: jugglerDef.properties,
+    settings: jugglerDef.options,
+  });
+}
+
+// TODO: move this function to '@loopback/repository'
+function defineModelClass<T extends typeof Model>(
+  base: T /* Model or Entity */,
+  definition: ModelDefinition,
+): T {
+  const modelName = definition.name;
+  const defineNamedModelClass = new Function(
+    'BaseClass',
+    `return class ${modelName} extends BaseClass {}`,
+  );
+  const modelClass = defineNamedModelClass(base) as T;
+  assert.equal(modelClass.name, modelName);
+  modelClass.definition = definition;
+  console.log('Defined model class %s', definition.name);
+  return modelClass;
+}
+
+// This will be implemented by
+// https://github.com/strongloop/loopback-next/issues/2036
+// In particular https://github.com/strongloop/loopback-next/issues/3737
+function bootModelApi(
+  app: TodoListApplication,
+  dsName: string,
+  modelClass: typeof Entity,
+  options: CrudRestControllerOptions,
+) {
+  const RepositoryClass = defineCrudRepositoryClass(modelClass);
```

Comment:
I am thinking about moving `defineCrudRepositoryClass` to `@loopback/repository`. I see the function as very coupled to `DefaultCrudRepository` class, not adding anything opinionated or controversial on top of the named constructor.